src/role_carry.js

```diff
@@ -37,8 +37,8 @@ roles.carry.preMove = function(creep, directions) {
       var source = creep.room.memory.position.creep[targetId];
       // TODO better the position from the room memory
       if (source !== null) {
-        let returnCode = creep.moveTo(Game.getObjectById(targetId));
-        if (creep.pos.getRangeTo(Game.getObjectById(targetId)) > 1) {
+        let returnCode = creep.moveTo(source);
+        if (creep.pos.getRangeTo(source) > 1) {
```

Comment:
yeah, RCL>3 and misplaced ( you should rename title :)
I talk about fact that source variable is defined as : ` var source = creep.room.memory.position.creep[targetId];`
I think the issue is here because it search target in the same room as where creep is actually. If totalschaden tryed it in only one room, it should work but i think it will be more chaotic with more room.